app/controllers/auth_controller.rb

```diff
@@ -0,0 +1,48 @@
+require 'net/http'
+require 'uri'
+require 'json'
+
+class AuthController < ApplicationController
+    def index
+    end
+
+    def signup
+        first_name = params[:first_name]
+        last_name = params[:last_name]
+        email = params[:email]
+        password = params[:password]
+
+        new_user = User.new(first_name: first_name, last_name: last_name, email: email)
+        new_user.save
+
+        uri = URI("https://identitytoolkit.googleapis.com/v1/accounts:signUp?key=#{ENV['FIREBASE_API_KEY']}")
+        res = Net::HTTP.post_form(uri, 'email' => email, 'password' => password)
+        data = JSON.parse(res.body)
+
+        puts User.all
+
+        if data.member?('error')
+            render status: 404
+        else
+            render status: :created
+        end
+    end
+
+    def login
+        email = params[:email]
+        password = params[:password]
+
+        uri = URI("https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=#{ENV['FIREBASE_API_KEY']}")
+        res = Net::HTTP.post_form(uri, 'email' => email, 'password' => password)
+        data = JSON.parse(res.body)
+
+        if data.member?('error')
+            render status: 404
```

Comment:
Same issue. `404` does not apply here. There's again two cases of errors:

1. User does not pass in one of the required parameters in the body and we return a `400` meaning `Bad Request` before we call the firebase API to save time.
2. User passes in all arguments correctly but the credentials (username and pw) are invalid so we return `401` meaning `Unauthorized`.